fix(sentry): Upgrade Segment package and fix Axios vulnerability

xrutayisire opened the pull request:
## Context

- Completes DT-1780

## The Solution

- Remove deprecated package https://github.com/segmentio/analytics-node
- Use new package https://github.com/segmentio/analytics-next/tree/master/packages/node#readme
- Follow guide https://segment.com/docs/connections/sources/catalog/libraries/server/node/migration/

## Impact / Dependencies

- New Analytics class required this usage

<img width="1046" alt="Screenshot 2023-11-23 at 09 52 41" src="https://github.com/prismicio/slice-machine/assets/19946868/6b3ca789-0fb4-4d0f-aa4a-d4b34b697474">

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

xrutayisire committed:
fix(sentry): Upgrade Sentry package and fix Axios vulnerability

xrutayisire committed:
Publish

 - cimsirp@1.21.1-dev-fix-vuln-sentry.0
 - next-upgrade@1.21.1-dev-fix-vuln-sentry.0
 - sveltekit@1.21.1-dev-fix-vuln-sentry.0
 - @slicemachine/adapter-next@0.3.27-dev-fix-vuln-sentry.0
 - @slicemachine/adapter-nuxt@0.3.27-dev-fix-vuln-sentry.0
 - @slicemachine/adapter-nuxt2@0.3.27-dev-fix-vuln-sentry.0
 - @slicemachine/adapter-sveltekit@0.3.27-dev-fix-vuln-sentry.0
 - @slicemachine/init@2.9.5-dev-fix-vuln-sentry.0
 - @slicemachine/manager@0.17.1-dev-fix-vuln-sentry.0
 - @slicemachine/plugin-kit@0.4.27-dev-fix-vuln-sentry.0
 - slice-machine-ui@1.21.1-dev-fix-vuln-sentry.0
 - start-slicemachine@0.12.7-dev-fix-vuln-sentry.0

bapmrl commented on packages/manager/src/managers/telemetry/TelemetryManager.ts:
```diff
@@ -41,39 +41,47 @@ type TelemetryManagerContext = {
 
 function assertTelemetryInitialized(
-	segmentClient: SegmentClient | undefined,
+	segmentClient: (() => Analytics) | undefined,
```
💡 Would it be better to change all references to segment (and in this case, `segmentClient`) to `analytics`?

xrutayisire replied:
I thought exactly the same but before in the old package doc it was already `Analytics`, but they decided to name it `SegmentClient`, I guess for more consistency in our codebase. Before it was am unmanned export vs now a named one but I guess how we are writing it does not depend on that.

So for me, we can keep this name as it doesn't really depend on the package version or type. WDYT?

See: https://github.com/segmentio/analytics-node#usage

bapmrl commented on packages/manager/src/managers/telemetry/TelemetryManager.ts:
```diff
@@ -41,39 +41,47 @@ type TelemetryManagerContext = {
-			},
-		});
+				console.error(
+					`An error occurred during telemetry initialization`,
```
⚠️ It's a callback and errors might be thrown after the initialization.

xrutayisire replied:
That's how I want it 😛 If I understood your comment.

What I want is on any Segment error that can happen in the life of the client, I want to console.error. It's documented here: https://segment.com/docs/connections/sources/catalog/libraries/server/node/#error-handling

bapmrl commented on packages/manager/src/managers/telemetry/TelemetryManager.ts:
```diff
@@ -41,39 +41,47 @@ type TelemetryManagerContext = {
+				// Since it's a local app, we do not benefit from event batching the way a server would normally do, all tracking event will be awaited.
+				maxEventsInBatch: 1,
+				// TODO: Verify that this actually does not send data to Segment when false.
+				disable: isTelemetryEnabled === false,
```
💡 `!isTelemetryEnabled`.

xrutayisire replied:
In Slice Machine UI we even have a rule to prevent that usage, right?

I started since I'm at Prismic to do explicit comparison, so I guess we should fix how we want to do, no?

xrutayisire committed:
review

xrutayisire committed:
Publish

 - cimsirp@1.21.1-dev-fix-vuln-sentry.1
 - next-upgrade@1.21.1-dev-fix-vuln-sentry.1
 - sveltekit@1.21.1-dev-fix-vuln-sentry.1
 - @slicemachine/adapter-next@0.3.27-dev-fix-vuln-sentry.1
 - @slicemachine/adapter-nuxt@0.3.27-dev-fix-vuln-sentry.1
 - @slicemachine/adapter-nuxt2@0.3.27-dev-fix-vuln-sentry.1
 - @slicemachine/adapter-sveltekit@0.3.27-dev-fix-vuln-sentry.1
 - @slicemachine/init@2.9.5-dev-fix-vuln-sentry.1
 - @slicemachine/manager@0.17.1-dev-fix-vuln-sentry.1
 - @slicemachine/plugin-kit@0.4.27-dev-fix-vuln-sentry.1
 - slice-machine-ui@1.21.1-dev-fix-vuln-sentry.1
 - start-slicemachine@0.12.7-dev-fix-vuln-sentry.1

xrutayisire committed:
Merge branch 'dev-next-release' into dev-fix-vuln-sentry
